jira-slack-server-integration/jira-slack-server-integration-plugin/src/main/java/com/atlassian/jira/plugins/slack/web/rest/SlackMessageResource.java

```diff
@@ -273,6 +274,16 @@ public String getOriginalKey() {
                     public boolean isArchived() {
                         return false;
                     }
+
+                    @Override
+                    public ApplicationUser getArchivedBy() {
+                        return null;
+                    }
+
+                    @Override
+                    public Date getArchivedDate() {
+                        return null;
+                    }
```

Comment:
Could you please elaborate on why these `null` attributes were added here.

Reply:
`SlackMessageResource` class is responsible for rendering a notification preview during the Slack post-function configuration. It is implemented by creation of a dummy `JiraPostFunctionEvent` and passing it to the renderer, that extracts issue, project and other meta information from the event object. `Issue` and `Project` are interfaces, so to create a dummy instances of them anonymous sub-classes are created in `SlackMessageResource` specifically for preview rendering. Those anonymous classes must contain implementations of all interface methods to pass the compilation, but the actual values returned by the dummy methods aren't important. New methods `getArchivedBy()` and `getArchivedDate()` were added to the `Project` interface in Jira 9.x, so I had to add them to the class - anonymous implementation of `Project` interface - in order to make the code valid for compilation. `null`s returned by these methods represent a non-deleted project - a valid use case.